mitmproxy/proxy/protocol2/http.py

```diff
@@ -108,9 +112,9 @@ def read_response_headers(self):
             self.state = self.read_response_body
             yield from self.read_response_body()  # there may already be further events.
         elif isinstance(event, h11.InformationalResponse):
-            if event.status_code == 101:
-                # FIXME: check if this is actually WebSocket
-                child_layer = websocket.WebsocketLayer(self.context, None)
+            self.flow.response.headers = http.Headers(event.headers)
+            if event.status_code == 101 and websockets.check_handshake(self.flow.respone.headers):
```

Comment:
Oops..
There are no tests present for http layer yet. Also, since this is a mock object python doesn't complain about self.flow not having a respone attribute